app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Any:
+    memory = {}
+
+    def inner(*args) -> str:
```

Comment:
You could consider using a decorator from `functools` for `inner`.

Reply:
The return type of `str` inner could be not only `str`, actually it can be any type.